met-api/src/met_api/services/membership_service.py

```diff
@@ -17,44 +19,77 @@ class MembershipService:
     def create_membership(engagement_id, request_json: dict):
         """Create membership."""
         user_id = request_json.get('user_id')
-        user: StaffUserModel = StaffUserModel.get_user_by_external_id(user_id)
-        if not user:
+        user_db: StaffUserModel = StaffUserModel.get_user_by_external_id(user_id)
+        if not user_db:
             raise BusinessException(
                 error='Invalid User.',
                 status_code=HTTPStatus.BAD_REQUEST)
 
+        user = StaffUserSchema().dump(user_db)
+        # attach and map groups
+        StaffUserService.attach_groups([user])
         # this makes sure duplicate membership doesnt happen.
         # Can remove when user can have multiple roles with in same engagement.
-        MembershipService._validate_team_member(engagement_id, user)
-
-        membership = MembershipService._create_membership_model(engagement_id, user)
-
-        KEYCLOAK_SERVICE.add_user_to_group(user_id=user_id, group_name=KeycloakGroups.EAO_TEAM_MEMBER.name)
+        MembershipService._validate_member(engagement_id, user)
+        group_name, membership_type = MembershipService._get_membership_details(user)
+        MembershipService._add_user_group(user, group_name)
+        membership = MembershipService._create_membership_model(engagement_id, user, membership_type)
         membership.commit()
         return membership
 
     @staticmethod
-    def _validate_team_member(engagement_id, user):
-        existing_membership = MembershipModel.find_by_engagement_and_user_id(engagement_id, user.id)
-        if existing_membership:
+    def _get_membership_details(user: StaffUserModel):
+        group_name = Groups.EAO_TEAM_MEMBER.name
+        membership_type = MembershipType.TEAM_MEMBER
+
+        if Groups.EAO_REVIEWER.value in user.get('groups'):
+            group_name = Groups.EAO_REVIEWER.name
+            membership_type = MembershipType.REVIEWER
+        return group_name, membership_type
```

Comment:
can you write some comments here?
not very evident on the logic!
the method name is _get_membership_details ;but feels like we might need to explain the logic in the if statement..
would be helpful to know whats the method is doing

Reply:
i came back to this method and re-read..i think it really needs an explanation ..
sorry to be picky on this one

Reply:
Great, thank you. Refactored can you please recheck